features/steps/permissions_component.go

```diff
@@ -0,0 +1,154 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	componenttest "github.com/ONSdigital/dp-component-test"
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	dpMongoDriver "github.com/ONSdigital/dp-mongodb/v2/mongodb"
+	"github.com/ONSdigital/dp-permissions-api/api"
+	"github.com/ONSdigital/dp-permissions-api/config"
+	"github.com/ONSdigital/dp-permissions-api/mongo"
+	"github.com/ONSdigital/dp-permissions-api/service"
+	serviceMock "github.com/ONSdigital/dp-permissions-api/service/mock"
+	"github.com/benweissmann/memongo"
+	"github.com/cucumber/godog"
+	"github.com/gofrs/uuid"
+	"go.mongodb.org/mongo-driver/bson"
+	"net/http"
+)
+
+type PermissionsComponent struct {
+	ErrorFeature   componenttest.ErrorFeature
+	svc            *service.Service
+	errorChan      chan error
+	MongoClient    *mongo.Mongo
+	Config         *config.Config
+	HTTPServer     *http.Server
+	ServiceRunning bool
+}
+
+func NewPermissionsComponent(mongoFeature *componenttest.MongoFeature) (*PermissionsComponent, error) {
+
+	f := &PermissionsComponent{
+		HTTPServer:     &http.Server{},
+		errorChan:      make(chan error),
+		ServiceRunning: false,
+	}
+
+	var err error
+
+	f.Config, err = config.Get()
+	if err != nil {
+		return nil, err
+	}
+
+	getMongoURI := fmt.Sprintf("localhost:%d", mongoFeature.Server.Port())
+	databaseName := memongo.RandomDatabase()
+
+	f.Config.MongoConfig.Database = databaseName
+	f.Config.MongoConfig.BindAddr = getMongoURI
+	f.Config.MongoConfig.Username, f.Config.MongoConfig.Password = createCredsInDB(getMongoURI, databaseName)
+
+	mongodb := &mongo.Mongo{}
+
+	if err := mongodb.Init(f.Config.MongoConfig); err != nil {
+		return nil, err
+	}
+
+	f.MongoClient = mongodb
+
+	return f, nil
+}
+
+func createCredsInDB(getMongoURI string, databaseName string) (string, string) {
+	username := "admin"
+	password, _ := uuid.NewV4()
+	mongoConnectionConfig := &dpMongoDriver.MongoConnectionConfig{
+		IsSSL:                   false,
+		ConnectTimeoutInSeconds: 15,
+		QueryTimeoutInSeconds:   15,
+
+		Username:        "",
+		Password:        "",
+		ClusterEndpoint: getMongoURI,
+		Database:        databaseName,
+	}
+	mongoConnection, err := dpMongoDriver.Open(mongoConnectionConfig)
+	if err != nil {
+		panic("expected db connection to be opened")
```

Comment:
I think you should return this error and let the calling code decide what to do with it.

Reply:
Updated PR to return error